core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -205,175 +204,163 @@ class FastPassService(protected val ctx: RawlsRequestContext,
-      return DBIO.successful()
+      logger.debug(s"FastPass is disabled. Will not grant FastPass access to ${workspace.toWorkspaceName}")
+      return Future.successful()
     }
+    dataSource
+      .inTransaction { implicit dataAccess =>
+        for {
+          rawlsServiceAccountUserInfo <- DBIO.from(googleServicesDAO.getServiceAccountUserInfo())
+          maybeSamUserInfo <- DBIO.from(
+            samDAO.getUserIdInfo(email, RawlsRequestContext(rawlsServiceAccountUserInfo)).map {
+              case SamDAO.NotFound =>
+                logger.warn(s"$email not found in Sam. Cannot setup FastPass.")
+                None
+              case SamDAO.NotUser =>
+                logger.warn(s"$email is not a user. Might be a group. Cannot setup FastPass.")
+                None
+              case SamDAO.User(userIdInfo) => Some(SamUserInfo.fromSamUserIdInfo(userIdInfo))
+            }
+          )
+          if maybeSamUserInfo.isDefined
+          samUserInfo = maybeSamUserInfo.orNull
+          _ = logger.info(s"Syncing FastPass grants for $email in ${workspace.toWorkspaceName}")
 
-    try
-      DBIO.from(quotaAvailableForClonedWorkspaceFastPassGrants(parentWorkspace, childWorkspace)).flatMap {
-        quotaAvailable =>
-          if (quotaAvailable) {
-            logger.info(
-              s"Adding FastPass access for ${ctx.userInfo.userEmail} in workspace being cloned ${parentWorkspace.toWorkspaceName}"
-            )
-            val expirationDate = DateTime.now(DateTimeZone.UTC).plus(config.grantPeriod.toMillis)
-            for {
-              maybeUserStatus <- DBIO.from(samDAO.getUserStatus(ctx))
-              if maybeUserStatus.isDefined
-              samUserInfo = maybeUserStatus.map(SamUserInfo.fromSamUserStatus).orNull
-
-              petEmail <- DBIO.from(samDAO.getUserPetServiceAccount(ctx, childWorkspace.googleProjectId))
-
-              userType = getUserType(samUserInfo.userEmail)
-              userAndPet = UserAndPetEmails(samUserInfo.userEmail, userType, petEmail)
-              _ <- removeParentBucketReaderGrant(parentWorkspace, samUserInfo)
-              _ <- setupBucketRoles(parentWorkspace,
-                                    Set(SamWorkspaceRoles.reader),
-                                    userAndPet,
-                                    samUserInfo,
-                                    expirationDate
-              )
-              _ <- DBIO
-                .from(
-                  openTelemetry.incrementCounter("fastpass-granted-user", tags = openTelemetryTags).unsafeToFuture()
-                )
-            } yield ()
-          } else {
-            logger.info(
-              s"Not enough IAM Policy Role Binding quota available to add FastPass access for ${ctx.userInfo.userEmail.value} in parent workspace ${parentWorkspace.toWorkspaceName}"
+          _ <- removeFastPassesForUserInWorkspace(workspace, samUserInfo)
```

Comment:
Just to make sure I understand this -- during the sync we remove any grants they have and then add the ones they should have?

Reply:
Yup!